core/src/main/scala/com/avast/clients/storage/StorageException.scala

```diff
@@ -7,4 +7,7 @@ object StorageException {
   case class InvalidResponseException(status: Int, body: String, desc: String, cause: Throwable = null)
       extends StorageException(s"Invalid response with status $status: $desc", cause)
 
+  case class InvalidDataException(status: Int, body: String, desc: String, cause: Throwable = null)
+    extends StorageException(s"Invalid response with status $status: $desc", cause)
```

Comment:
Maybe change the message? Sth like "Invalid response data..."

Reply:
Makes sense, change to: "Invalid response data [ with status $status ]: $desc"